app/models/maintenance_stats/queries/github/releases_query.rb

```diff
@@ -0,0 +1,77 @@
+module MaintenanceStats
+  module Queries
+    module Github
+      class RepoReleasesQuery < BaseQuery
+        RELEASES_QUERY = Rails.application.config.graphql.client.parse <<-GRAPHQL
+          query($owner: String!, $repo_name: String!, $cursor: String){
+            repository(owner: $owner, name: $repo_name){
+              releases(first: 100, after: $cursor, orderBy: {field: CREATED_AT, direction: DESC}){
+                nodes {
+                  name
+                  publishedAt
+                }
+                totalCount
+                pageInfo {
+                  hasPreviousPage
+                  hasNextPage
+                  endCursor
+                  startCursor
+                }
+              }
+            }
+          }
+        GRAPHQL
+
+        VALID_PARAMS = [:owner, :repo_name, :end_date]
+        REQUIRED_PARAMS = [:owner, :repo_name]
+
+        def self.client_type
+          :v4
+        end
+
+        def query(params: {})
+          validate_params(params)
+
+          end_date = params[:end_date]
+          params = params.slice!(:end_date)
+
+          releases = nil
+
+          has_next_page = true
+          cursor = nil
+          while(has_next_page)
+            params[:cursor] = cursor if cursor.present?
+            result = @client.query(RELEASES_QUERY, variables: params)
+
+            # send back the releases we have so far if we encounter an error
+            return releases if check_for_errors(result)
+
+            has_next_page = result.data.repository.releases.page_info.has_next_page
+            cursor = result.data.repository.releases.page_info.end_cursor
+
+            # initalize releases if we have not started gathering data
+            releases ||= []
```

Comment:
On line 38 `releases = []` would make this unnecessary, I think? Unless it's important to return nil?

Reply:
I'm using `nil` to indicate no data was returned from the API. At this line we have a response from GitHub so I want to initialize to an empty array unless this is after the first page of information.